File: exercism-python/hamming/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("same length strands are required.")

    # diff = 0
    # for i in range(len(strand_a)):
    #     if strand_a[i] != strand_b[i]:
    #         diff += 1
    
    # return diff

    return sum(a != b for a, b in zip(strand_a, strand_b))

File: exercism-go/hamming/hamming.go
// Package hamming implements function to calculate the Hamming difference.
package hamming

import "errors"

// Distance calculates the Hamming difference between two DNA strands
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("same length strands are required")
	}

	diff := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			diff++
		}
	}

	return diff, nil
}
